apps/remix-ide/src/app/udapp/run-tab.js

```diff
@@ -44,6 +44,43 @@ export class RunTab extends LibraryPlugin {
     this.compilersArtefacts = compilersArtefacts
     this.networkModule = networkModule
     this.fileProvider = fileProvider
+    this.setupEvents()
+  }
+
+  setupEvents () {
+    this.blockchain.events.on('newTransaction', (tx, receipt) => {
+      this.emit('newTransaction', tx, receipt)
+    })
+  }
+
+  getSettings () {
+    return new Promise((resolve, reject) => {
+      if (!this.container) reject(new Error('UI not ready'))
+      else {
+        resolve({
+          selectedAccount: this.container.querySelector('#txorigin').selectedOptions[0].value,
```

Comment:
could you replace `.container.querySelector('#txorigin')` by call to `this.settingsUI`

Reply:
same for the others settings